base_classes/NXsource.nxdl.xml

```diff
@@ -163,7 +163,41 @@
 		<doc>For storage rings, the current at the end of the most recent injection.</doc>
 		<attribute name="time" type="NX_DATE_TIME"><doc>date and time of the most recent injection.</doc></attribute>
 	</field>
-        <group name="geometry" type="NXgeometry" deprecated="Use the field `depends_on` and :ref:`NXtransformations` to position the source and NXoff_geometry to describe its shape instead">
+	<field name="photon_energy" type="NX_FLOAT" units="NX_ENERGY">
```

Comment:
source is a source of radiation not necessarily photons only, next person needs electron_energy and generates again a term which is arguably "needed".

See "type" which sources of radiation adding an energy and stating in its docstring that it is the energy of the radiation specified by source will make the story work equally well. @rettigl  this is one example why likely such parts were taken out before the code camp by @sanbrock

Reply:
I agree this maybe does not make so much sense. Maybe we did use NXsource outside its intended usecase